File: SportFixtures/SportFixtures.Data/Entities/Sport.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SportFixtures.Data.Entities
{
    public class Sport
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public ICollection<Team> Teams { get; set; }

        public override bool Equals(object obj)
        {
            bool equals = false;

            if (obj != null && this.GetType() == obj.GetType())
            {
                Sport sport = (Sport)obj;
                equals = sport.Name.Equals(Name);
            }

            return equals;
        }

        public Sport()
        {
            Teams = new List<Team>();
        }

    }
}

File: SportFixturesFront/src/app/shared/models/sport.ts
import { Team } from "./team";
import { EncounterMode } from "./encounterMode";

export class Sport {
    public id: string;
    public name: string;
    public encounterMode: EncounterMode;
    public teams: Team[];

    constructor() {
        this.name = "";
        this.encounterMode = EncounterMode.Double;
        this.teams = [];
    }
}